### PR TITLE
Create associations

### PR DESCRIPTION
**Pull request by Kaiserabbas**
# Creating the remaining classes for our school library and creating the associations between them.
- Created a class Classroom with the following:
  - @label instance variable, should be initialized in the constructor.
  - Setter and getter for @label (remember about attr_accessor).
- Created the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
  - Created the has-many side (a classroom has many students).
  - Created the belongs-to side (a student belongs to a classroom).
  - Made sure that when adding a student to a classroom it also sets the classroom for the student.
  - Made sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Created a class Book with the following:
  - @title and @author instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about attr_accessor).
- Created a class Rental with the following:
  - @date instance variable, should be initialized in the constructor.
  - Setter and getter for @date (remember about attr_accessor).
- Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
  - Created the has-many side of Book and Rental (a book has many rentals).
  - Created the belongs-to side of Rental and Book (a rental belongs to a book).
  - Created the has-many side of Person and Rental (a person has many rentals).
  - Created the belongs-to side of Rental and Person (a rental belongs to a person).
  - Modified the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: src/person.rb [hunk 1]
```diff
@@ -13,10 +13,15 @@ def initialize(age, name: 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def add_rental(date, person)
+    Rental.new(date, self, person)
   end
```
Comment (OyePriscilla) [inline src/person.rb]: - [ ] Please kindly take note that the `self` keyword typically refers to the current instance of the class where the code is located (in this case `Person`); I think you will be more accurate to pass the `book` parameter as part of the arguement as stated in the requirement and remove `person`. ℹ️ 

project requirement

- [ ] Kindly ensure that   *Getters and Setters for @rentals is present, using getters and setters in your classes provides a level of control over attribute access. ℹ️

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Kaiserabbas 👍 👍 👍 👍 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Implemented a program using the command line. ✔️
Set up associations between classes and objects. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/book.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(date, book)
+    Rental.new(date, self, book)
```
Comment (zeff96) [inline src/book.rb]: - [ ] Please kindly take note that the self keyword typically refers to the current instance of the class where the code is located (in this case Book); I think you will be more accurate to pass the `person` parameter as part of the argument as stated in the requirement and remove `book`.
- [ ] Kindly ensure that *Getters and Setters for https://github.com/rentals is present, using getters and setters in your classes provides a level of control over attribute access. ℹ️

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️ ♻️ ♻️ 
Hi @Kaiserabbas , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

close

 ## Highlights
✔️ Linters passing
✔️ Descriptive pull request

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Johnadibe — APPROVED** [review]
# Status: Approved ✔️
### Hi @Kaiserabbas👋🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

LeonardoInGatsby

## Highlights
- Implemented all changes requested by the previous reviewer✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**